### ISSUE TITLE
[bugprone-narrowing-conversions] False positive for `bool` to `char` conversion

### ISSUE THREAD

**Issue by chengfeng-xie** [llvm/llvm-project#191337]
For the following code ([CE](https://godbolt.org/z/YsTo5rhbq)):

```cpp
void test(bool b) {
  auto c = char{b};
}
```

Clang-Tidy warns:

```cpp
<source>:2:17: warning: narrowing conversion from 'bool' to signed type 'char' is implementation-defined [bugprone-narrowing-conversions]
    2 |   auto c = char{b};
      |                 ^
```

However, for this conversion, the standard mandates ([[conv.integral]/2](https://eel.is/c++draft/conv.integral#2.sentence-2)):

> [...] If the source type is `bool`, the value `false` is converted to zero and the value `true` is converted to one.

So no narrowing occurs here.
